skbio/io/format/stockholm.py

```diff
@@ -577,3 +576,79 @@ def _check_for_malformed_line(line, expected_len):
         raise StockholmFormatError('Line contains %d item(s). It must '
                                    'contain exactly %d item(s).'
                                    % (len(line), expected_len))
+
+
+@stockholm.writer(TabularMSA)
+def _tabular_msa_to_stockholm(obj, fh):
+    index = list(obj.index)
+    # Writes header
+    fh.write("# STOCKHOLM 1.0\n")
+
+    # Writes GF data to file
+    if obj.has_metadata():
+        for i in range(len(obj.metadata)):
```

Comment:
This works but can be rewritten to be more efficient and Pythonic by iterating over the keys and values in the dictionary instead of indexing (we use `viewitems` instead of `.items()` to be efficient for both Python 2 and 3):

```python
from future.utils import viewitems
for feature_name, feature_data in viewitems(obj.metadata):
    ...
```